File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/presentation/ui/utils/PopUpActivity.java
package com.lattis.ellipse.presentation.ui.utils;

import android.app.Activity;
import android.content.Intent;
import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import android.view.View;

import com.lattis.ellipse.presentation.ui.base.activity.BaseCloseActivity;
import com.lattis.ellipse.presentation.view.CustomTextView;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.OnClick;
import io.lattis.ellipse.R;

public class PopUpActivity extends BaseCloseActivity<PopUpActivityPresenter> implements PopUpActivityView {

    public final static String TITLE_POP_UP = "TITLE_POP_UP";
    public final static String SUBTITLE1_POP_UP = "SUBTITLE1_POP_UP";
    public final static String SUBTITLE2_POP_UP = "SUBTITLE2_POP_UP";
    public final static String ACTIONBTN_POP_UP = "ACTIONBTN_POP_UP";
    public final static String ACTIONBTN_POP_UP2 = "ACTIONBTN_POP_UP2";


    public static void launchForResult(Activity activity, int requestCode, String title, String subTitle1, String subTitle2, String actionBtn) {
        Intent intent = new Intent(activity, PopUpActivity.class);
        intent.putExtra(TITLE_POP_UP,title);
        intent.putExtra(SUBTITLE1_POP_UP,subTitle1);
        intent.putExtra(SUBTITLE2_POP_UP,subTitle2);
        intent.putExtra(ACTIONBTN_POP_UP,actionBtn);
        activity.startActivityForResult(intent,requestCode);
    }

    public static void launchForResultFromFragment(Fragment fragment,Activity activity, int requestCode, String title, String subTitle1, String subTitle2, String actionBtn) {
        Intent intent = new Intent(activity, PopUpActivity.class);
        intent.putExtra(TITLE_POP_UP,title);
        intent.putExtra(SUBTITLE1_POP_UP,subTitle1);
        intent.putExtra(SUBTITLE2_POP_UP,subTitle2);
        intent.putExtra(ACTIONBTN_POP_UP,actionBtn);
        fragment.startActivityForResult(intent,requestCode);
    }


    @BindView(R.id.cv_title_pop_up)
    CustomTextView cv_title_pop_up;

    @BindView(R.id.cv_subtitle1_pop_up)
    CustomTextView cv_subtitle1_pop_up;

    @BindView(R.id.cv_subtitle2_pop_up)
    CustomTextView cv_subtitle2_pop_up;

    @BindView(R.id.cv_action_btn_pop_up)
    CustomTextView cv_action_btn_pop_up;


    @OnClick(R.id.iv_close_pop_up)
    public void actionCancel(){
        setResult(RESULT_CANCELED);
        finish();
    }

    @OnClick(R.id.cv_action_btn_pop_up)
    public void actionOK(){
        setResult(RESULT_OK);
        finish();
    }


    @Inject
    PopUpActivityPresenter presenter;

    @NonNull
    @Override
    protected PopUpActivityPresenter getPresenter() {
        return presenter;
    }

    @Override
    protected int getActivityLayoutId() {
        return R.layout.activity_pop_up;
    }

    @Override
    protected void inject() {
        getComponent().inject(this);
    }

    @Override
    public void setView(String title, String subTitle1, String subTitle2, String actionBtn) {
        if(title!=null){
            cv_title_pop_up.setText(title);
        }else{
            cv_title_pop_up.setVisibility(View.INVISIBLE);
        }

        if(subTitle1!=null){
            cv_subtitle1_pop_up.setText(subTitle1);
        }else{
            cv_subtitle1_pop_up.setVisibility(View.INVISIBLE);
        }

        if(subTitle2!=null && !subTitle2.equals("")){
            cv_subtitle2_pop_up.setText(subTitle2);
        }else{
            cv_subtitle2_pop_up.setVisibility(View.GONE);
        }

        if(actionBtn!=null){
            cv_action_btn_pop_up.setText(actionBtn);
        }else{
            cv_action_btn_pop_up.setVisibility(View.INVISIBLE);
        }
    }

    @Override
    protected void onInternetConnectionChanged(boolean isConnected) {

    }
}

File: velolabs_repos/lattis_android_v2/app/src/main/java/com/lattis/lattis/presentation/popup/PopUpActivity.kt
package com.lattis.lattis.presentation.popup

import android.app.Activity
import android.content.Intent
import android.text.TextUtils
import android.view.View
import androidx.fragment.app.Fragment
import com.lattis.lattis.presentation.authentication.signin.SignInActivityPresenter
import com.lattis.lattis.presentation.authentication.signin.SignInActivityView
import com.lattis.lattis.presentation.ui.base.activity.BaseActivity
import io.lattis.lattis.R
import kotlinx.android.synthetic.main.activity_pop_up.*
import javax.inject.Inject

class PopUpActivity : BaseActivity<PopUpActivityPresenter, PopUpActivityView>(),
    PopUpActivityView {


    @Inject
    override lateinit var presenter: PopUpActivityPresenter
    override val activityLayoutId = R.layout.activity_pop_up
    override var view: PopUpActivityView = this

    override fun configureViews() {
        super.configureViews()
        btn_pop_up_positive1.setOnClickListener {
            var intent = Intent()
            intent.putExtra(POSITIVE_LEVEL,1)
            setResult(Activity.RESULT_OK,intent)
            finish()
        }

        btn_pop_up_positive2.setOnClickListener {
            intent.putExtra(POSITIVE_LEVEL,2)
            setResult(Activity.RESULT_OK,intent)
            finish()
        }

        btn_pop_up_positive3.setOnClickListener {
            intent.putExtra(POSITIVE_LEVEL,3)
            setResult(Activity.RESULT_OK,intent)
            finish()
        }

        btn_pop_up_negative.setOnClickListener {
            setResult(RESULT_CANCELED)
            finish()
        }
    }

    override fun setView(
        title: String?,
        subTitle1: String?,
        subTitle2: String?,
        actionBtnPositive1: String?,
        actionBtnPositive2: String?,
        actionBtnPositive3: String?,
        actionBtnNegative: String?
    ) {
        if (!TextUtils.isEmpty(title)) {
            pop_up_title.text = (title)
        } else {
            pop_up_title.setVisibility(View.GONE)
        }
        if (!TextUtils.isEmpty(subTitle1)) {
            pop_up_sub_title1.setText(subTitle1)
        } else {
            pop_up_sub_title1.setVisibility(View.GONE)
        }

        if (!TextUtils.isEmpty(subTitle2)) {
            pop_up_sub_title2.setText(subTitle2)
        } else {
            pop_up_sub_title2.setVisibility(View.GONE)
        }

        if (!TextUtils.isEmpty(actionBtnPositive1)) {
            btn_pop_up_positive1.setText(actionBtnPositive1)
        } else {
            btn_pop_up_positive1.setVisibility(View.GONE)
        }

        if (!TextUtils.isEmpty(actionBtnPositive2)) {
            btn_pop_up_positive2.setText(actionBtnPositive2)
        } else {
            btn_pop_up_positive2.setVisibility(View.GONE)
        }

        if (!TextUtils.isEmpty(actionBtnPositive3)) {
            btn_pop_up_positive3.setText(actionBtnPositive3)
        } else {
            btn_pop_up_positive3.setVisibility(View.GONE)
        }

        if (!TextUtils.isEmpty(actionBtnNegative)) {
            btn_pop_up_negative.setText(actionBtnNegative)
        } else {
            pop_up_divider.visibility=View.GONE
            btn_pop_up_negative.setVisibility(View.GONE)
        }
    }

    protected override fun onInternetConnectionChanged(isConnected: Boolean) {}

    companion object {
        const val TITLE_POP_UP = "TITLE_POP_UP"
        const val SUBTITLE1_POP_UP = "SUBTITLE1_POP_UP"
        const val SUBTITLE2_POP_UP = "SUBTITLE2_POP_UP"
        const val ACTIONBTN_POSITIVE1 = "ACTIONBTN_POSITIVE1"
        const val ACTIONBTN_POSITIVE2 = "ACTIONBTN_POSITIVE2"
        const val ACTIONBTN_POSITIVE3 = "ACTIONBTN_POSITIVE3"
        const val ACTIONBTN_NEGATIVE = "ACTIONBTN_NEGATIVE"
        const val POSITIVE_LEVEL = "POSITIVE_LEVEL"



        fun launchForResult(
            activity: Activity,
            requestCode: Int,
            title: String?,
            subTitle1: String?,
            subTitle2: String?,
            actionBtnPositive1: String?,
            actionBtnPositive2: String?,
            actionBtnPositive3: String?,
            actionBtnNegative: String?
        ) {
            val intent = Intent(activity, PopUpActivity::class.java)
            intent.putExtra(TITLE_POP_UP, title)
            intent.putExtra(SUBTITLE1_POP_UP, subTitle1)
            intent.putExtra(SUBTITLE2_POP_UP, subTitle2)
            intent.putExtra(ACTIONBTN_POSITIVE1, actionBtnPositive1)
            intent.putExtra(ACTIONBTN_POSITIVE2, actionBtnPositive2)
            intent.putExtra(ACTIONBTN_POSITIVE3, actionBtnPositive3)
            intent.putExtra(ACTIONBTN_NEGATIVE, actionBtnNegative)
            activity.startActivityForResult(intent, requestCode)
        }

        fun launchForResultFromFragment(
            fragment: Fragment,
            activity: Activity?,
            requestCode: Int,
            title: String?,
            subTitle1: String?,
            subTitle2: String?,
            actionBtnPositive1: String?,
            actionBtnPositive2: String?,
            actionBtnPositive3: String?,
            actionBtnNegative: String?
        ) {
            val intent = Intent(activity, PopUpActivity::class.java)
            intent.putExtra(TITLE_POP_UP, title)
            intent.putExtra(SUBTITLE1_POP_UP, subTitle1)
            intent.putExtra(SUBTITLE2_POP_UP, subTitle2)
            intent.putExtra(ACTIONBTN_POSITIVE1, actionBtnPositive1)
            intent.putExtra(ACTIONBTN_POSITIVE2, actionBtnPositive2)
            intent.putExtra(ACTIONBTN_POSITIVE3, actionBtnPositive3)
            intent.putExtra(ACTIONBTN_NEGATIVE, actionBtnNegative)
            fragment.startActivityForResult(intent, requestCode)
        }
    }


}